File: AAS.ScheduleMaker/Classes/SubjectCoursePair.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AAS.ScheduleMaker
{
    public class SubjectCoursePair
    {
        public string Subject { get; set; }
        public int CourseNumber { get; set; }
        public SubjectCoursePair(string _subject, int _courseNumber)
        {
            Subject = _subject.ToUpper();
            CourseNumber = _courseNumber;
        }

        public static bool operator ==(SubjectCoursePair a, SubjectCoursePair b)
        {
            return a.CourseNumber == b.CourseNumber && a.Subject.ToUpper() == b.Subject.ToUpper();
        }

        public static bool operator !=(SubjectCoursePair a, SubjectCoursePair b)
        {
            return a.CourseNumber != b.CourseNumber || a.Subject.ToUpper() != b.Subject.ToUpper();
        }
    }
}

File: AggieAutoSchedulerAngular/ClientApp/src/app/classes/subject-course-pair.ts
export class SubjectCoursePair {
  subject: string;
  courseNumber: number;
  crns: number[];
  constructor(subject?: string, courseNumber?: number, crns?: number[]) {
    this.subject = subject;
    this.courseNumber = courseNumber;
    this.crns = crns;
  }
}
